db/impl/etcd/client.go

```diff
@@ -317,3 +318,141 @@ func (c *Client) ListPrefix(prefix string, obj db.Entity) ([]db.Entity, error) {
 
 	return c.traverse(resp.Node, obj), nil
 }
+
+// Acquire and permanently hold a lock. Attempts until timeout. If timeout is
+// zero, it will only try once.
+func (c *Client) Acquire(lock db.Lock) error {
+	logrus.Debugf("Acquiring lock %v", lock)
+
+	content, err := jsonio.Write(lock)
+	if err != nil {
+		return errors.LockFailed.Combine(err)
+	}
+
+	path, err := lock.Path()
+	if err != nil {
+		return errors.LockFailed.Combine(err)
+	}
+
+	_, err = c.client.Set(context.Background(), c.qualified(path), string(content), &client.SetOptions{PrevExist: client.PrevNoExist})
+	if er, ok := err.(client.Error); ok && er.Code == 105 {
```

Comment:
The constant in our version of the library has the wrong code.

I'll comment this.

-Erik

On 24 Aug 2016, at 9:30, Bill Robinson wrote:

>> +// zero, it will only try once.
>> +func (c *Client) Acquire(lock db.Lock) error {
>> +	logrus.Debugf("Acquiring lock %v", lock)
>> +
>> +	content, err := jsonio.Write(lock)
>> +	if err != nil {
>> +		return errors.LockFailed.Combine(err)
>> +	}
>> +
>> +	path, err := lock.Path()
>> +	if err != nil {
>> +		return errors.LockFailed.Combine(err)
>> +	}
>> +
>> +	_, err = c.client.Set(context.Background(), c.qualified(path), 
>> string(content), &client.SetOptions{PrevExist: client.PrevNoExist})
>> +	if er, ok := err.(client.Error); ok && er.Code == 105 {
>
> What is 105 here?  Can we add a constant for this?
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/444/files/2b3726db59c59297cc9d671a722b2fb161d35f37#r76091383